voorhoede/tudelft-spacefinder
# Feat/filter by occupancy

# Changes

- There's now some naive logic for calculating occupancy in `lib\calculate-occupancy.ts`. We may fine-tune it or make it configurable per-building later
- We temporarily do not deal with space/room occupancy in any way. All places in the UI, including space-related, display occupancy of the building (e.g. where the space resides)
- The OccupancyIndicator component is now used everywhere to display occupancy
- Everything related to opening hours is now back in the UI to check how it looks together with the occupancy indicator. It will be hidden by a feature flag in another PR
- The filter menu now has an option to filter by occupancy. The map still has its separate filter logic working. When updated building occupancy data appears, the dataset of the map is set anew

# Associated issue

Resolves https://trello.com/c/KUxxegFN/17-filter-buildings-by-occupancy
Moves to doing (?) https://trello.com/c/ozxRuXGu/22-configure-occupancy-thresholds-for-all-spaces-rule

# How to test

1. Make sure that per-building occupancy is displayed for building in list, building details, space in overall list, space details. The tooltip for the indicator currently displays the number of active devices for facilitating testing and fine-tuning
2. Make sure that spaces in lists and on the map are correctly filtered by building occupancy
3. Make sure that if the occupancy filter is applied when new building occupancy data appears and changes the value of the indicator for some spaces, it immediately affects not only the displayed indicators but also the filtered set of spaces (can be done updating device count in the buildings table in the DB)

# Checklist
